ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/folsom/MemcachedClientBuilder.java

```diff
@@ -85,11 +89,27 @@ private MemcachedClientBuilder(final Transcoder<T> transcoder) {
   }
 
   public MemcachedClientBuilder<T> withObjectSizeMonitoring(final MetricFactory metricFactory, final String cacheName, final long sampleRate) {
-    transcoder = new ObjectSizeMonitoringTranscoder<>(transcoder, metricFactory, cacheName, sampleRate);
+    objectSizeMonitoringApplier = (transcoder) -> new ObjectSizeMonitoringTranscoder<>(transcoder, metricFactory, cacheName, sampleRate);
+    return this;
+  }
+
+  public MemcachedClientBuilder<T> withCompression() {
+    compressionApplier = (transcoder) -> new CompressionTranscoder<T>(transcoder, CompressionAlgorithm.LZ4.getInstance());
```

Comment:
CompressionAlgorithm can have 'default' method returning default impl, and this will just call it